+ source ../common/net_lib.sh
++ source /root/mugen/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n function ']'
+ config_params
+ LOG_INFO 'Start to config params of the case.'
+ message='Start to config params of the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to config params of the case.'
Fri Apr 28 09:56:56 2023 - INFO  - Start to config params of the case.
+ get_free_eth 2
+ local num_eth=2
+ echo eth0
eth0
++ python3 /root/mugen/libs/locallibs/get_test_device.py --node 1 --device nic
Fri Apr 28 09:57:05 2023 - INFO  - pkgs:(lshw) is already installed
+ NODE1_NIC='eth1 eth2 eth3'
+ LOCAL_ETH=(${NODE1_NIC[@]/$(ip route | grep ${NODE1_IPV4} | awk '{print$3}')/})
++ ip route
++ grep 10.198.114.2
++ awk '{print$3}'
+ '[' 3 -ge 2 ']'
+ test_eth1=eth1
+ test_eth2=eth2
+ LOG_INFO 'End to config params of the case.'
+ message='End to config params of the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to config params of the case.'
Fri Apr 28 09:57:14 2023 - INFO  - End to config params of the case.
++ type -t pre_test
+ '[' -n '' ']'
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run test.'
+ message='Start to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to run test.'
Fri Apr 28 09:57:15 2023 - INFO  - Start to run test.
+ ip link add dev blue type vrf table 1001
Error: Unknown device type.
+ CHECK_RESULT 2
+ actual_result=2
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 2 ']'
+ '[' 0 -eq 0 ']'
+ test 2x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 32'
+ message='oe_test_net_VRF.sh line 32'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 32'
Fri Apr 28 09:57:16 2023 - ERROR - oe_test_net_VRF.sh line 32
+ '[' 0 -eq 1 ']'
+ return 0
+ find /sys/class/net/
+ grep blue
+ CHECK_RESULT 1
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 34'
+ message='oe_test_net_VRF.sh line 34'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 34'
Fri Apr 28 09:57:16 2023 - ERROR - oe_test_net_VRF.sh line 34
+ '[' 0 -eq 1 ']'
+ return 0
+ ip link set dev blue up
Cannot find device "blue"
+ CHECK_RESULT 1
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 36'
+ message='oe_test_net_VRF.sh line 36'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 36'
Fri Apr 28 09:57:17 2023 - ERROR - oe_test_net_VRF.sh line 36
+ '[' 0 -eq 1 ']'
+ return 0
+ ip link set dev eth1 master blue
Error: argument "blue" is wrong: Device does not exist

+ ip link set dev eth1 up
+ CHECK_RESULT 0
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ ip addr add dev eth1 192.0.2.1/24
+ ip addr show
+ grep blue
+ CHECK_RESULT 1
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 42'
+ message='oe_test_net_VRF.sh line 42'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 42'
Fri Apr 28 09:57:18 2023 - ERROR - oe_test_net_VRF.sh line 42
+ '[' 0 -eq 1 ']'
+ return 0
+ ip link add dev red type vrf table 1002
Error: Unknown device type.
+ CHECK_RESULT 2
+ actual_result=2
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 2 ']'
+ '[' 0 -eq 0 ']'
+ test 2x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 45'
+ message='oe_test_net_VRF.sh line 45'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 45'
Fri Apr 28 09:57:19 2023 - ERROR - oe_test_net_VRF.sh line 45
+ '[' 0 -eq 1 ']'
+ return 0
+ ip link set dev red up
Cannot find device "red"
+ CHECK_RESULT 1
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 47'
+ message='oe_test_net_VRF.sh line 47'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 47'
Fri Apr 28 09:57:20 2023 - ERROR - oe_test_net_VRF.sh line 47
+ '[' 0 -eq 1 ']'
+ return 0
+ ip link set dev eth2 master red
Error: argument "red" is wrong: Device does not exist

+ ip link set dev eth2 up
+ CHECK_RESULT 0
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ ip addr add dev eth2 192.0.2.1/24
+ ip addr show
+ grep ' red'
+ CHECK_RESULT 1
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log=
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n ''
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_net_VRF.sh line 53'
+ message='oe_test_net_VRF.sh line 53'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_net_VRF.sh line 53'
Fri Apr 28 09:57:21 2023 - ERROR - oe_test_net_VRF.sh line 53
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'End to run test.'
+ message='End to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to run test.'
Fri Apr 28 09:57:22 2023 - INFO  - End to run test.
+ CASE_RESULT 0
+ case_re=0
+ test -z 7
+ test 7 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Fri Apr 28 09:57:23 2023 - ERROR - failed to execute the case.
+ exit 7
+ post_test
+ LOG_INFO 'Start to restore the test environment.'
+ message='Start to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to restore the test environment.'
Fri Apr 28 09:57:23 2023 - INFO  - Start to restore the test environment.
+ ip link del blue
Cannot find device "blue"
+ ip link del red
Cannot find device "red"
+ nmcli con delete eth1 eth2
Error: unknown connection 'eth1'.
Connection 'eth2' (40ddb5ff-2a02-44f9-bc30-50de64100d50) successfully deleted.
Error: cannot delete unknown connection(s): 'eth1'.
+ LOG_INFO 'End to restore the test environment.'
+ message='End to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to restore the test environment.'
Fri Apr 28 09:57:25 2023 - INFO  - End to restore the test environment.
